docs/development/release.md

```diff
@@ -0,0 +1,9 @@
+# Release Process
```

Comment:
@chrislovecnm Can we nuke this in favor of https://github.com/kubernetes/kops/pull/1053